dashboard/app/reporting_email.go

```diff
@@ -110,7 +110,9 @@ func (cfg *EmailConfig) Validate() error {
 // Assuming it is called June 15, the monthly report will cover April-May diff.
 func handleCoverageReports(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	curHostPort := r.URL.Host
+	if coverageDBClient != nil { // initialized in prod deployment, nil in tests
+		ctx = SetCoverageDBClient(r.Context(), coverageDBClient)
```

Comment:
(Unrelated to this PR)

It does look like you'd better pass this DB client explicitly rather than hiding it inside the Context.